treeherder/changelog/collector.py

```diff
@@ -0,0 +1,97 @@
+""" Collector, grabs changes in various sources and put them in a DB.
+"""
+import json
+import os
+
+import github3
+from django.conf import settings
+
+from treeherder.changelog.filters import Filters
+
+MAX_ITEMS = 100
+CFG = os.path.join(os.path.dirname(__file__), "repositories.json")
+
+with open(CFG) as f:
+    CFG = json.loads(f.read())
+
+
+class GitHub:
+    def __init__(self):
+        self.token = os.environ["GITHUB_CLIENT_SECRET"]
+        self.id = os.environ["GITHUB_CLIENT_ID"]
+        self.gh = github3.login(token=self.token, two_factor_callback=self._2FA)
+        self.filters = Filters()
+
+    def _2FA(self):
+        """ The two-factor authentication is useful when running Github from a
+        personal github account.
+        """
+        if "localhost" not in settings.SITE_HOSTNAME:
+            raise ValueError("Cannot perform 2FA if not running locally")
```

Comment:
I don't know that we need to raise a ValueError here. I was thinking there would be a check for localhost before this function is even called (if localhost, call _2FA otherwise skip).